tests/modules/images/regenerate-existing-images/background-process/perflab-background-job-test.php

```diff
@@ -0,0 +1,228 @@
+		$this->assertEquals( 1, $attempts );
+	}
+
+	/**
+	 * @covers ::should_run
+	 */
+	public function test_job_should_run_for_job() {
+		$this->job = perflab_create_background_job( 'test' );
+
+		$run = $this->job->should_run();
+		$this->assertTrue( $run );
+	}
+
+	/**
+	 * @covers ::should_run
+	 */
+	public function test_job_should_not_run_for_completed_job() {
+		$this->job = perflab_create_background_job( 'test' );
+
+		// Mark job as complete.
+		$this->job->set_status( 'perflab_job_complete' );
+		$run = $this->job->should_run();
+
+		$this->assertFalse( $run );
+	}
+
+	/**
+	 * @covers ::lock
+	 * @covers ::unlock
+	 */
+	public function test_lock_unlock() {
+		$this->job = perflab_create_background_job( 'test' );
+		$time      = time();
+		$this->job->lock( $time );
+		$lock_time = get_term_meta( $this->job->get_id(), 'perflab_job_lock', true );
+		$this->assertSame( absint( $lock_time ), $time );
+		$this->job->unlock();
+		$lock_time = get_term_meta( $this->job->get_id(), 'perflab_job_lock', true );
+		$this->assertEmpty( $lock_time );
+	}
+
+	/**
+	 * @covers ::get_start_time
+	 */
+	public function test_get_start_time() {
+		$this->job = perflab_create_background_job( 'test' );
+		$time      = time();
+		$this->job->lock( $time );
+		$lock_time = get_term_meta( $this->job->get_id(), 'perflab_job_lock', true );
+		$this->assertSame( absint( $lock_time ), $time );
+
+		$start_time = $this->job->get_start_time();
+		$this->assertEquals( $start_time, $lock_time );
+		$this->assertEquals( $start_time, $time );
+		$this->assertEquals( $time, $lock_time );
+	}
+
+	public function test_max_attempts_filter() {
+		$job = perflab_create_background_job( 'test' );
+		add_filter( 'perflab_job_max_attempts_allowed', array( $this, 'max_attempts_10' ) );
+		$job->should_run();
+		remove_filter( 'perflab_job_max_attempts_allowed', array( $this, 'max_attempts_10' ) );
+		$filter_ran = property_exists( $this, 'attempt_filter_ran' );
+
+		$this->assertTrue( $filter_ran );
+	}
+
+	public function test_max_attempts_limit_reached() {
+		$job                   = perflab_create_background_job( 'test' );
+		$before_attempts_limit = $job->should_run();
+		update_term_meta( $job->get_id(), $job::META_KEY_JOB_ATTEMPTS, 2 );
+		add_filter( 'perflab_job_max_attempts_allowed', array( $this, 'max_attempts_1' ) );
+		$after_attempts_limit = $job->should_run();
+		remove_filter( 'perflab_job_max_attempts_allowed', array( $this, 'max_attempts_1' ) );
+
+		$this->assertTrue( $before_attempts_limit );
+		$this->assertFalse( $after_attempts_limit );
+	}
+
+	public function max_attempts_10() {
+		$this->attempt_filter_ran = true;
+
+		return 10;
+	}
+
+	public function max_attempts_1() {
+		return 1;
+	}
+
+	/**
+	 * Runs after each test.
+	 *
+	 * @return void
+	 */
+	public function tear_down() {
+		if ( $this->job instanceof Perflab_Background_Job ) {
+			wp_delete_term( $this->job->get_id(), 'background_job' );
+		}
+	}
```

Comment:
See above, this method should not be necessary, as the WordPress test suite handles this automatically.